python-for-android/dists/kolibri/src/main/java/org/learningequality/Task.java

```diff
@@ -188,7 +188,12 @@ public static void updateProgress(
             String id, String notificationTitle, String notificationText, int progress, int total
     ) {
         NotificationRef ref = Worker.buildNotificationRef(id);
-        Manager manager = new Manager(ContextUtil.getApplicationContext(), ref);
-        manager.send(notificationTitle, notificationText, progress, total);
+        try {
+            Context context = ContextUtil.getApplicationContext();
```

Comment:
If this was where the context issue was originating from then it probably has to do with removing the worker reference from this utility in combination with using some non-`RemoteListenableWorker` workers. Since the non-`RemoteListenableWorker` workers would be started by the system, we wouldn't necessarily have access to the foreground service. Storing static instances of the worker is bad, not only for the reasons regarding the context, but also because multiple instances could be running at once, so you don't necessarily get the instance you expect.

An alternative would be to go back to having all workers be `RemoteListenableWorker`s. Or use a sort of context manager approach in the worker such that this utility returns that context when called within that block

Reply:
I'm going to merge this regardless. I added a follow up issue